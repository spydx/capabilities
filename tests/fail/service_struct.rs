use capabilities_derive::service;

#[service]
struct Database {
    pub _db: String,
}

#[allow(dead_code)]
fn main() {

}