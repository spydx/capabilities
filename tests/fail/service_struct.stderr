error: Unsupported literal, has to be a type from Capabilities lib
 --> $DIR/service_struct.rs:9:1
  |
9 | #[service]
  | ^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `service` (in Nightly builds, run with -Z macro-backtrace for more info)

error: custom attribute panicked
 --> $DIR/service_struct.rs:9:1
  |
9 | #[service]
  | ^^^^^^^^^^
  |
  = help: message: called `Option::unwrap()` on a `None` value
